conn_test.go

```
@@ -40,20 +44,80 @@ func (m *mockStream) Read(p []byte) (int, error) {
 	<-ctx.Done()
 	return nil, ctx.Err()
 }
 
+func TestCapsuleWritesAreCoalesced(t *testing.T) {
+	writes := make(chan []byte)
+	writeStarted := make(chan struct{})
+	conn := newProxiedConn(&mockStream{
+		writeStarted: writeStarted,
+		written:      writes,
+	})
+	t.Cleanup(func() { conn.Close() })
```

golangci-lint (2.13.2)

[error] Error return value of `conn.Close` is not checked

(errcheck)